tests/automated/microlibrary/fault_reporting_output_stream/main.cc

```diff
@@ -0,0 +1,411 @@
+
+    auto const string = "UStwlrJdq";
+
+    EXPECT_CALL( stream.driver(), put( string ) ).WillOnce( Return( Result<void>{} ) );
+
+    EXPECT_FALSE( stream.put( string ).is_error() );
+
+    EXPECT_TRUE( stream.is_nominal() );
+}
+
+/**
+ * \brief Verify microlibrary::Fault_Reporting_Output_Stream::put( std::uint8_t ) properly
+ *        handles a put error.
+ */
+TEST( putDataErrorHandling, putError )
+{
+    auto stream = Mock_Fault_Reporting_Output_Stream{};
+
+    auto const error = Mock_Error{ 9 };
+
+    EXPECT_CALL( stream.driver(), put( A<std::uint8_t>() ) ).WillOnce( Return( error ) );
+
+    auto const result = stream.put( std::uint8_t{ 0xB6 } );
+
+    EXPECT_TRUE( result.is_error() );
+    EXPECT_EQ( result.error(), error );
+
+    EXPECT_FALSE( stream.end_of_file_reached() );
+    EXPECT_FALSE( stream.io_error_reported() );
+    EXPECT_TRUE( stream.fatal_error_reported() );
+}
+
+/**
+ * \brief Verify microlibrary::Fault_Reporting_Output_Stream::put( std::uint8_t ) works
+ *        properly.
+ */
+TEST( putData, worksProperly )
+{
+    auto stream = Mock_Fault_Reporting_Output_Stream{};
+
+    auto const data = std::uint8_t{ 0xFB };
+
+    EXPECT_CALL( stream.driver(), put( SafeMatcherCast<std::uint8_t>( Eq( data ) ) ) )
+        .WillOnce( Return( Result<void>{} ) );
+
+    EXPECT_FALSE( stream.put( data ).is_error() );
+
+    EXPECT_TRUE( stream.is_nominal() );
+}
+
+/**
+ * \brief Verify microlibrary::Fault_Reporting_Output_Stream::put( std::uint8_t const *,
+ *        std::uint8_t const * ) properly handles a put error.
+ */
+TEST( putDataBlockErrorHandling, putError )
+{
+    auto stream = Mock_Fault_Reporting_Output_Stream{};
+
+    auto const error = Mock_Error{ 101 };
+
+    EXPECT_CALL( stream.driver(), put( A<std::vector<std::uint8_t>>() ) ).WillOnce( Return( error ) );
+
+    auto const data = std::vector<std::uint8_t>{ 0x04, 0x35, 0x13, 0x53 };
+
+    auto const result = stream.put( &*data.begin(), &*data.end() );
+
+    EXPECT_TRUE( result.is_error() );
+    EXPECT_EQ( result.error(), error );
+
+    EXPECT_FALSE( stream.end_of_file_reached() );
+    EXPECT_FALSE( stream.io_error_reported() );
+    EXPECT_TRUE( stream.fatal_error_reported() );
+}
+
+/**
+ * \brief Verify microlibrary::Fault_Reporting_Output_Stream::put( std::uint8_t const *,
+ *        std::uint8_t const * ) works properly.
+ */
+TEST( putDataBlock, worksProperly )
+{
+    auto stream = Mock_Fault_Reporting_Output_Stream{};
+
+    auto const data = std::vector<std::uint8_t>{ 0x49, 0x70, 0xB9, 0x97, 0x02 };
+
+    EXPECT_CALL( stream.driver(), put( data ) ).WillOnce( Return( Result<void>{} ) );
+
+    EXPECT_FALSE( stream.put( &*data.begin(), &*data.end() ).is_error() );
+
+    EXPECT_TRUE( stream.is_nominal() );
+}
+
+/**
+ * \brief Verify microlibrary::Fault_Reporting_Output_Stream::print() properly handles a
+ *        microlibrary::Output_Formatter::print() error.
+ */
+TEST( printErrorHandling, outputFormatterPrintError )
```

Comment:
This test needs to be reworked.
1. The fatal error flag should only be set if a stream I/O driver data sink write function reports an error.
2. Pass multiple values to the print call, have the first output formatter print call fail, and verify the second output formatter print call does not occur.